### PR TITLE
[SPARK-5456] [SQL] fix decimal compare for jdbc rdd

### PR DESCRIPTION
**Pull request by adrian-wang**

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala [hunk 1]
```diff
@@ -271,8 +271,11 @@ class JDBCSuite extends FunSuite with BeforeAndAfter {
     assert(rows(0).getDouble(0) === 1.00000000000000022) // Yes, I meant ==.
     assert(rows(0).getDouble(1) === 1.00000011920928955) // Yes, I meant ==.
     assert(rows(0).getAs[BigDecimal](2)
-        .equals(new BigDecimal("123456789012345.54321543215432100000")))
+      .equals(new BigDecimal("123456789012345.54321543215432100000")))
     assert(rows(0).schema.fields(2).dataType === DecimalType(40, 20))
+    val compareDecimal = sql("SELECT C FROM flttypes where C > C - 1").collect()
+    assert(compareDecimal(0).getAs[BigDecimal](0)
+      .equals(new BigDecimal("123456789012345.54321543215432100000")))
```
Comment (yhuai) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala]: This case triggered the bug, right?

Comment (adrian-wang) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala]: yes.

**Comment by rxin** [comment]
@adrian-wang is this still up to date?

**Comment by rxin** [comment]
Alright I talked to @yhuai offline. Going to merge this.

**Comment by rxin** [comment]
I merged it in master, branch-1.4, and branch-1.3.

**Pull request closed without merging**